tensornetwork/contractors/stochastic_contractor.py

```diff
@@ -78,12 +78,18 @@ def contract_trace_edges(
   for node in initial_node_set:
     trace_edges, flag_none, total_dim = set(), False, 1
     new_node = node
-    for edge, dim in zip(node.edges, list(node.get_tensor().shape)):
-      if edge.node1 is edge.node2:
+    # makes sure node_edges points to the original edges
+    # even after contracting the trace
+    node_edges = [e for e in node.edges]
+    node_dims = list(node.get_tensor().shape)
+    for edge, dim in zip(node_edges, node_dims):
+      if (not edge.is_disabled) and (edge.node1 is edge.node2):
         if edge not in trace_edges:
           # Contract trace edge
-          new_node = net.contract(edge)
+          new_node = net.contract(edge, name=node.name)
```

Comment:
I see, so contracting a trace edge will name the "new" edge have the same name. I guess this is better than what we were doing before, though if someone sets the node `via contract(edge, name="...")`, we should default to doing that.

Reply:
Good point. I just changed _contract_trace to do the same thing.